### DISCUSSION TITLE
Workflow action

### DISCUSSION THREAD

**Discussion by ramanveerji** [nikhilbadyal/docker-py-revanced#374] [Q&A] [answered]
How to add scheduled cron job in build apk workflow so that it automatically builds apk.

**Comment by nikhilbadyal** [comment]
One of the user @Spacellary has forked this repo and added automated builds. Check it out [here](https://github.com/Spacellary/ReVanced-Extended-Automated-Builder)

**Reply by ramanveerji** [reply]
Thats a different thing. It checks for update as per the schedule and then pushes update. But what I am doing is, I don't want to check it for updates, I want the builder to run on selected day only.

**Reply by nikhilbadyal** [reply]
That should be much simpler. 

Explain the issue you are facing with your implementation and what have you tried so far.

**Reply by ramanveerji** [reply]
> That should be much simpler. 
> 
> Explain the issue you are facing with your implementation and what have you tried so far.

Check this [thing](https://github.com/ramanveerji/docker-py-revanced/blob/main/.github/workflows/build-apk.yml) line 44

**Reply by ramanveerji** [reply]
The template is not valid. .github/workflows/build-apk.yml (Line: 54, Col: 25): Unexpected value '',.github/workflows/build-apk.yml (Line: 55, Col: 22): Unexpected value ''

I don't know why this error popping up, that thing is unchanged.

**Comment by Spacellary** [comment, marked as answer]
@ramanveerji 

Does this work for you?

https://github.com/Spacellary/ReVanced-Extended-Automated-Builder/blob/main/.github/workflows/scheduled-build.yml

**Reply by ramanveerji** [reply]
I have added this file to my repo and for the time being I have set the cron job to run every 5 minutes, and it started roughly after 13-14 minutes. But it started finally. Thanks for this thing. Also, I have added your update check workflow, now I can use both things whichever is needed at the time.

**Reply by ramanveerji** [reply]
Thanks @nikhilbadyal bro for helping me and solving my query. 🤗

**Reply by Spacellary** [reply]
@ramanveerji It will try to run as close to every 5 min as possible but GitHub machines are always overused so that's why it runs at closer to every 20mins sometimes.

I advise you to run at most once every hour actually, spamming such a long build action doesn't sound good and you could be Rate Limited, instead try running at most every 1 hour or 12h and just use the update checker at every 5-20 mins to trigger builds earlier if needed instead 😅

**Reply by nikhilbadyal** [reply]
I have an opinion similar to @Spacellary . Don't take free service for granted. You might get limited or get a ban altogether for abusing service. Use wisely. 

Before running, always check if a run is even required. 
If you are building some apps whose latest versions are always supported. Along with the logic @Spacellary implemented(checking the patches version) you can parse the latest version of the app available using packages like [this](https://pypi.org/project/google-play-scraper/) and only run if version is higher than the last time it ran.

**Reply by Spacellary** [reply]
@nikhilbadyal I believe they already desabled it ([here](https://github.com/ramanveerji/docker-py-revanced/commit/3624a8cbe0c1c48e5f96ed26fc74b6b5e05e4d78)) but yea, it's best to risk ruining a good thing for everyone else for spam or abuse of their free resources. I only tracked patch versions because there's no point in updating just for a version update in most cases for me and that just means breaking patches, and because that way it's also a super simple check for what matters the most for me, being the patch features. But it could be improved a lot, specially looking at how much cleaner Eren is doing it -- [workflows/update.yml](https://github.com/IMXEren/rvx-builds/blob/main/.github/workflows/update.yml) and [scripts/check_updates.py](https://github.com/IMXEren/rvx-builds/blob/main/scripts/check_updates.py).

> If you are building some apps whose latest versions are always supported.

But yeah, that'd be a valid use case. Good suggestion.

**Reply by ramanveerji** [reply]
I was just testing the 5 minute thing. It was working. I have disabled that. @Spacellary

**Reply by ramanveerji** [reply]
@nikhilbadyal Absolutely i will not do that. Instead i will either run the schedule twice a month or check for patch update thrice a month.

**Reply by Spacellary** [reply]
At least the simple patch update checker workflow isn't really that bad, you can run that multiple times every week or daily without any issues at least.

**Reply by ramanveerji** [reply]
> At least the simple patch update checker workflow isn't really that bad, you can run that multiple times every week or daily without any issues at least.

That's good but suppose if by any chance revanced updated their patches in 4-5 days then it will build and release every 4-5 days. And I have to update very soon 😂. So, I will check the update on 10,20 and 30 of month, and if there will be an update then it will build.

**Reply by ramanveerji** [reply]
Guys when we enable tg upload why it doesn't show the patch Changelog? 

It shows this only. No hyperlink is there. Is there any other configuration to be done?

New Release(s)🥳
See Changelog here

**Reply by Spacellary** [reply]
Never used TG Upload, maybe @nikhilbadyal knows more.
